### PR TITLE
intl: improve coverage for Intl.NumberFormat

### PR DESCRIPTION
**Pull request by ryzokuken**
Add tests for Intl.NumberFormat.prototype[@@toStringTag].

/cc @gsathya @littledan @anba @rwaldron

### CODE REVIEW

File: test/intl402/NumberFormat/prototype/toStringTag/prop-desc.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-intl.numberformat.prototype-@@tostringtag
+description: >
+  Tests that Intl.NumberFormat.prototype[@@toStringTag] has the required attributes.
+includes: [propertyHelper.js]
+---*/
+
+verifyProperty(Intl.NumberFormat, Symbol.toStringTag, {
```
Comment (gsathya) [inline test/intl402/NumberFormat/prototype/toStringTag/prop-desc.js]: This should be checking Intl.NumerFormat.prototype

Comment (gsathya) [reply test/intl402/NumberFormat/prototype/toStringTag/prop-desc.js]: This should be checking the value as well

File: test/intl402/NumberFormat/prototype/toStringTag/initial-value.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright (C) 2018 Ujjwal Sharma. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-intl.numberformat.prototype-@@tostringtag
+description: >
+  Initial value of Intl.NumberFormat.prototype[@@toStringTag] is "Object".
+---*/
+
+assert.sameValue(Intl.NumberFormat.prototype[Symbol.toStringTag], 'Object');
```
Comment (gsathya) [inline test/intl402/NumberFormat/prototype/toStringTag/initial-value.js]: nit: is the formatting off here?

File: test/intl402/NumberFormat/prototype/toStringTag/initial-value.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+assert.sameValue(
+  Object.prototype.toString.call(new Intl.NumberFormat()),
+  '[object Object]'
+);
```
Comment (gsathya) [inline test/intl402/NumberFormat/prototype/toStringTag/initial-value.js]: Add tests that reconfigure the toStringTag and assert that's reflected with both a regular lookup and in a toString call similar to the above two tests

File: test/intl402/NumberFormat/prototype/toStringTag/prop-desc.js (hunk 1 above)
Comment (ryzokuken) [reply test/intl402/NumberFormat/prototype/toStringTag/prop-desc.js]: Oops, on it.

File: test/intl402/NumberFormat/prototype/toStringTag/initial-value.js (hunk 2 above)
Comment (ryzokuken) [reply test/intl402/NumberFormat/prototype/toStringTag/initial-value.js]: Do you mean this wrt the line width?

File: test/intl402/NumberFormat/prototype/toStringTag/prop-desc.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+esid: sec-intl.numberformat.prototype-@@tostringtag
+description: >
+  Tests that Intl.NumberFormat.prototype[@@toStringTag] has the required attributes.
+includes: [propertyHelper.js]
+---*/
+
+verifyProperty(Intl.NumberFormat.prototype, Symbol.toStringTag, {
+  writable: false,
+  enumerable: false,
+  configurable: true
```
Comment (gsathya) [inline test/intl402/NumberFormat/prototype/toStringTag/prop-desc.js]: Still need to verify the `value` prop

File: test/intl402/NumberFormat/prototype/toStringTag/configurable.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2018 Ujjwal Sharma. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-intl.numberformat.prototype-@@tostringtag
+description: >
+  Initial value of Intl.NumberFormat.prototype[@@toStringTag] is "Object".
```
Comment (gsathya) [inline test/intl402/NumberFormat/prototype/toStringTag/configurable.js]: This is no longer an accurate description of the test.

**Pull request merged**